contracts/sol6/katana/KatanaFeeHandler.sol

```diff
@@ -0,0 +1,459 @@
+    address public kyberNetwork;
+    address public feePool;
+    IERC20 public immutable knc;
+
+    uint256 public immutable burnBlockInterval;
+    uint256 public lastBurnBlock;
+
+    BRRData public brrData;
+
+    /// @dev amount of eth to burn for each burn knc call
+    uint256 public weiToBurn = 2 ether;
+
+    mapping(address => uint256) public feePerPlatformWallet;
+    mapping(address => uint256) public rebatePerWallet;
+    // total balance in the contract that is for rebate, reward, platform fee
+    uint256 public totalPayoutBalance;
+    /// @dev use to get rate of KNC/ETH to check if rate to burn knc is normal
+    /// @dev index 0 is currently used contract address, indexes > 0 are older versions
+    ISanityRate[] internal sanityRateContract;
+
+    event FeeDistributed(
+        IERC20 indexed token,
+        address indexed platformWallet,
+        uint256 platformFeeWei,
+        uint256 rewardWei,
+        uint256 rebateWei,
+        address[] rebateWallets,
+        uint256[] rebatePercentBpsPerWallet,
+        uint256 burnAmtWei
+    );
+
+    event BRRUpdated(uint256 rewardBps, uint256 rebateBps, uint256 burnBps);
+
+    event FeePoolUpdated(address feePool);
+
+    event RebatePaid(address indexed rebateWallet, IERC20 indexed token, uint256 amount);
+
+    event PlatformFeePaid(address indexed platformWallet, IERC20 indexed token, uint256 amount);
+
+    event KncBurned(uint256 kncTWei, IERC20 indexed token, uint256 amount);
+
+    event EthReceived(uint256 amount);
+
+    event BurnConfigSet(ISanityRate sanityRate, uint256 weiToBurn);
+
+    event KyberNetworkUpdated(address kyberNetwork);
+
+    event KyberProxyUpdated(IKyberProxy kyberProxy);
+
+    constructor(
+        IKyberProxy _kyberProxy,
+        address _kyberNetwork,
+        IERC20 _knc,
+        uint256 _burnBlockInterval,
+        address _daoOperator,
+        address _feePool
+    ) public DaoOperator(_daoOperator) {
+        require(_kyberProxy != IKyberProxy(0), "kyberNetworkProxy 0");
+        require(_kyberNetwork != address(0), "kyberNetwork 0");
+        require(_knc != IERC20(0), "knc 0");
+        require(_burnBlockInterval != 0, "_burnBlockInterval 0");
+        require(_feePool != address(0), "feePool 0");
+
+        kyberProxy = _kyberProxy;
+        kyberNetwork = _kyberNetwork;
+        knc = _knc;
+        burnBlockInterval = _burnBlockInterval;
+        feePool = _feePool;
+
+        // set default brrData
+        brrData.rewardBps = uint16(DEFAULT_REWARD_BPS);
+        brrData.rebateBps = uint16(DEFAULT_REBATE_BPS);
+    }
+
+    modifier onlyKyberNetwork {
+        require(msg.sender == address(kyberNetwork), "only kyberNetwork");
+        _;
+    }
+
+    modifier onlyNonContract {
+        require(tx.origin == msg.sender, "only non-contract");
+        _;
+    }
+
+    receive() external payable {
+        emit EthReceived(msg.value);
+    }
+
+    /// @dev only call by daoOperator
+    function setBRRData(
```

Comment:
can change to use PermissionAdmin instead of daoOperator, otherwise we will need to add transfer dao operator function to this contract

Reply:
Assume in the first stage it will be the current daoOperator, but can be transferred to Governance's executor or vice versa

Reply:
As discuss, I added another function to transfer daoOperator